<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>900</width><height>600</height></rect>
  </property>
  <property name="windowTitle">
   <string>Dashboard</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">
QPushButton {
    background-color: #2e89ff;
    color: white;
    border-radius: 6px;
    padding: 6px 12px;
    font: bold 10pt "Segoe UI";
}
QPushButton:hover {
    background-color: #1c6edb;
}
QLabel#welcomeLabel {
    font: bold 14pt "Segoe UI";
}
QTableView {
    border: 1px solid #ccc;
    gridline-color: lightgray;
    font: 10pt "Segoe UI";
    selection-background-color: #2e89ff;
}
    </string>
   </property>

   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,1,0">
    <!-- Welcome Label -->
    <item>
     <widget class="QLabel" name="welcomeLabel">
      <property name="text">
       <string>Welcome, User [Role]</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>

    <!-- Attendance Table -->
    <item>
     <widget class="QTableView" name="attendanceTableView"/>
    </item>

    <!-- Chart Container -->
    <item>
     <widget class="QWidget" name="attendanceChartContainer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>

    <!-- Button Layout -->
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="registerUserButton">
        <property name="text">
         <string>Register New User</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exportCsvButton">
        <property name="text">
         <string>Export Attendance</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="markAttendanceButton">
        <property name="text">
         <string>Mark Attendance</string>
        </property>
        <property name="visible">
         <bool>false</bool>
        </property>
       </widget>
      </item>

      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation"><enum>Qt::Horizontal</enum></property>
        <property name="sizeType"><enum>QSizePolicy::Expanding</enum></property>
        <property name="sizeHint" stdset="0">
         <size><width>0</width><height>0</height></size>
        </property>
       </spacer>
      </item>

      <item>
       <widget class="QPushButton" name="updateProfileButton">
        <property name="text"><string>Update Profile</string></property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="themeToggleButton">
        <property name="text"><string>Change Theme</string></property>
       </widget>
      </item>
      <item>
	<widget class="QPushButton" name="backupRestoreButton">
 	<property name="text">
  	<string>Backup &amp; Restore</string>
 	</property>
	</widget>
      </item>
      <item>
       <widget class="QPushButton" name="logoutButton">
        <property name="text"><string>Logout</string></property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
